### PR TITLE
Convert the streaming server to ESM

### PR DESCRIPTION
**Pull request by renchap**
This is supported by Node 16+

Some notes:
- `__dirname` does not work in ES modules, but you can get the same result
- `delete session` was a no-op, because `delete` can only operate on object properties. Using ESM forces strict-mode, which fails at parsing on this rather than silently ignoring the statement. Added a comment to fix this in a future refactor.
- eslint does not support yet ESM configs (this will be supported in eslint 9), so the file has been kept as cjs

### CODE REVIEW

File: streaming/.eslintrc.cjs [hunk 1]
```diff
@@ -1,4 +1,8 @@
+/* eslint-disable import/no-commonjs */
+
 // @ts-check
+
+// @ts-ignore - This needs to be a CJS file (eslint does not yet support ESM configd), and TS is complaining we use require
```
Comment (ThisIsMissEm) [inline streaming/.eslintrc.cjs:5]: Maybe just remove the ts-check?

File: streaming/index.js [hunk 2]
```diff
@@ -1,30 +1,31 @@
...
+import { logger, httpLogger, initializeLogLevel, attachWebsocketHttpLogger, createWebsocketLogger } from './logging.js';
+import { setupMetrics } from './metrics.js';
+import { isTruthy, normalizeHashtag, firstParam } from './utils.js';
 
 const environment = process.env.NODE_ENV || 'development';
 
 // Correctly detect and load .env or .env.production file based on environment:
 const dotenvFile = environment === 'production' ? '.env.production' : '.env';
 
+const __dirname = url.fileURLToPath(new URL('.', import.meta.url));
```
Comment (ThisIsMissEm) [inline streaming/index.js:27]: Could we do '..' here and remove the path.join?

File: streaming/index.js [hunk 3]
```diff
@@ -143,7 +144,7 @@ const pgConfigFromEnv = (env) => {
   let baseConfig = {};
 
   if (env.DATABASE_URL) {
-    const parsedUrl = dbUrlToConfig(env.DATABASE_URL);
+    const parsedUrl = pgConnectionString.parse(env.DATABASE_URL);
```
Comment (ThisIsMissEm) [inline streaming/index.js:147]: Did they release the update? See the fix me PR below

**Review by ThisIsMissEm — COMMENTED** [review]
A few comments

File: streaming/index.js (hunk 3 above)
Comment (renchap) [reply streaming/index.js:147]: No, this has not been merged

File: streaming/.eslintrc.cjs (hunk 1 above)
Comment (renchap) [reply streaming/.eslintrc.cjs:5]: It is still useful so you get typecheck on `defineConfig`

File: streaming/index.js (hunk 2 above)
Comment (renchap) [reply streaming/index.js]: Good point, fixed :)

**Review by ykzts — DISMISSED** [review]

File: streaming/.eslintrc.cjs (hunk 1 above)
Comment (ClearlyClaire) [inline streaming/.eslintrc.cjs:5]: typo?
```suggestion
// @ts-ignore - This needs to be a CJS file (eslint does not yet support ESM configs), and TS is complaining we use require
```

File: streaming/index.js [hunk 4]
```diff
@@ -1482,13 +1486,15 @@ const startServer = async () => {
...
       // references to the websocket, the request, and the logger, causing
       // memory leaks.
-      //
-      // @ts-ignore
-      delete session;
+
+      // This is commented out because `delete` only operated on object properties
+      // It needs to be replaced by `session = undefined`, but it requires every calls to
+      // `session` to check for it, thus a significant refactor
+      // delete session;
```
Comment (ClearlyClaire) [inline streaming/index.js:1497]: Are we sure there is no such memory leak?

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**